### ISSUE TITLE
macros defining integer constants using expressions containing enum values don't work

### ISSUE THREAD

**Issue by heycam** [servo/rust-bindgen#258]
In Gecko (in gfxFontConstants.h) we have:

```C++
enum {
    eFeatureAlternates_historical,
    eFeatureAlternates_stylistic,
    eFeatureAlternates_styleset,
    eFeatureAlternates_character_variant,
    eFeatureAlternates_swash,
    eFeatureAlternates_ornaments,
    eFeatureAlternates_annotation,

    eFeatureAlternates_numFeatures
};

#define NS_FONT_VARIANT_ALTERNATES_HISTORICAL        (1 << eFeatureAlternates_historical)
```

rust-bindgen doesn't generate anything for NS_FONT_VARIANT_ALTERNATES_HISTORICAL.

**Comment by heycam** [comment]
cc @emilio

**Comment by emilio** [comment]
Ugh, this never worked though... I know how to fix it for this case, but it's not pretty, and it's definitely hard to make it work for the general case.

**Comment by emilio** [comment]
If only clang gave us a way of evaluating an expression in a given context...

**Comment by emilio** [comment]
Yeah, so I looked into patching libclang so it works, but I can't see a clear way to do this. Basically, clang needs to expand the macro so it knows in what context to evaluate it. You can convert that into a constant, and then I think I can do stuff with it.

**Comment by emilio** [comment]
FWIW, if those are switched to constants, they'll work properly with #260, even though it's a complex expression.

**Comment by pheki** [comment]
Just to point out other uses / problems, the same underlying problem happen for "function aliases" defines, such as [some functions in monocypher](https://github.com/LoupVaillant/Monocypher/blob/f2aaa1b4ebdd751bbcd29c74c4ec4cfa0cae67ec/src/optional/monocypher-ed25519.h#L131).
